csrc/ec_gen.cpp

```diff
@@ -159,12 +159,15 @@ JNIEXPORT void JNICALL Java_com_amazon_corretto_crypto_provider_EcGen_generateEc
   jbyteArray yArr,
   jbyteArray sArr)
 {
-    // TODO, figure out how to do this with EVP
     std::vector<uint8_t, SecureAlloc<uint8_t> > derBuf;
-    EC_KEY* ecParams = NULL;
+    EC_KEY *ecParams = NULL;
     EC_KEY *key = NULL;
     const EC_GROUP* group;
 
+    EVP_PKEY_CTX *keyCtx = NULL;
+    EVP_PKEY_CTX *tmpCtx = NULL;
+    EVP_PKEY *pkey = NULL;
```

Comment:
When possible we prefer to use [EvpKeyContext](https://github.com/corretto/amazon-corretto-crypto-provider/blob/fd967b00839fad0651d4cae77625cb5ff85d9896/csrc/keyutils.h#L14) which allows us to track these structures on the stack and know that they will be automatically freed as soon as the tracking object goes out of scope. This will allow you to remove all of the free statements except for `EC_KEY_free()` at the bottom.

Also, if you were to rename `tmpCtx` to `paramCtx`, it would be much clearer.

So I'd probably do:
```suggestion
EvpKeyContext keyCtx;
EvpKeyContext paramCtx;
```